src/components/eip3643/admin/DeployToken.tsx

```diff
@@ -106,16 +114,49 @@ export default function DeployToken({ onClose = () => {} }) {
   });
 
   useEffect(() => {
-    if (form.values.nftAddress && ethers.isAddress(form.values.nftAddress)) {
-      const requiresNftModuleCall = new ethers.Interface([
-        "function requireNFT(address _nftAddress)",
-      ]).encodeFunctionData("requireNFT", [form.values.nftAddress]);
-      form.setFieldValue("complianceSettings", [requiresNftModuleCall]);
+    if (complianceModuleSelected === requiresNftModuleAddress) {
+      if (form.values.nftAddress && ethers.isAddress(form.values.nftAddress)) {
+        const requiresNftModuleCall = new ethers.Interface([
+          "function requireNFT(address _nftAddress)",
+        ]).encodeFunctionData("requireNFT", [form.values.nftAddress]);
+        form.setFieldValue("complianceSettings", [requiresNftModuleCall]);
+      }
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [form.values.nftAddress]);
 
-  const [complianceModuleSelected, setComplianceModuleSelected] = useState("");
+  useEffect(() => {
+    if (complianceModuleSelected === maxOwnershipByCountryModuleAddress) {
+      try {
+        const maxOwnershipCountryModuleCall = new ethers.Interface([
+          "function setMaxPercentage(uint16 _country, uint16 _maxLocal, uint16 _maxNonlocal)",
+        ]).encodeFunctionData("setMaxPercentage", [
+          Number(form.values.maxOwnershipCountryCode),
+          Number(form.values.maxOwnershipCountryLocalPercentage),
+          Number(form.values.maxOwnershipCountryNonLocalPercentage),
+        ]);
+
+        form.setFieldValue("complianceSettings", [
+          maxOwnershipCountryModuleCall,
+        ]);
+
+        setComplianceError("");
+      } catch (e) {
+        console.error(e);
```

Comment:
Is it allowed to use `console`?

Reply:
I felt it was a fine middle ground between throwing an error and hiding it completely, so we show user friendly msg, but still show more details about the error in the console.